PerformanceTuning.md

```diff
@@ -0,0 +1,147 @@
+#KNN Performance Tuning
+
+
+In this section we provide recommendations for performance tuning to improve indexing/search performance with the k-NN plugin.  On a high level k-NN works on following principles
+
+* Graphs are created per (Lucene) segment
+* Queries execute on segments sequentially inside the shard (same as any other Elasticsearch query). 
+* Each graph in the segment returns ‘k’ neighbors and the size results with the highest score is returned by the coordinating node. Note that size can be greater or smaller than k.
+
+To improve performance it is necessary to keep the number of segments under control. Ideally having 1 segment per shard will give the optimal performance with respect to search latency. We can achieve more parallelism by having more shards per index. We can control the number of segments either during indexing by asking Elasticsearch to slow down the segment creation by disabling the refresh interval or choosing larger refresh interval, increasing the flush threshold OR force-merging to 1 segment after all the indexing finishes and before searches.
+
+##Indexing Performance Tuning
+
+Following steps could help improve indexing performance especially when you plan to index large number of vectors at once. 
+
+* Disable refresh interval  (Default = 1 sec)
+ 
+ Disable refresh interval or set a long duration for refresh interval to avoid creating multiple smaller segments
+  ```  
+    PUT /<index_name>/_settings
+        {
+            "index" : {
+                "refresh_interval" : "-1"
+            }
+        }
+  ```
+* Disable flush
+ ```
+    Increase the "index.translog.flush_threshold_size" to some bigger value lets say "10gb", default is 512MB
+ ```
+* No Replicas (No Elasticsearch replica shard)
+ ```
+    Having replication set to 0, will avoid duplicate construction of graphs in 
+    both primary and replicas. When we enable replicas after the indexing, the 
```

Comment:
Yes. I pointed the [doc](https://www.elastic.co/guide/en/elasticsearch/reference/master/tune-for-indexing-speed.html#_disable_replicas_for_initial_loads) to read more about this section. It is covered there. To avoid confusion adding a note in the paragraph.